### PR TITLE
Passed kata 1 test, code review time

### PR DESCRIPTION
**Pull request by Raothorn**
Not super great test coverage, probably need to test empty arrays and such.

### CODE REVIEW

File: go/kata-01/gaston/kata1.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+)
+
+func binarySearch(target int, list []int) int {
+	return getIndex(target, list, 0, len(list))
+}
+
+func getIndex(target int, list []int, start int, end int) int {
+	midpoint := (start + end) / 2
+
+	if start >= end {
```
Comment (JonesDavey) [inline go/kata-01/gaston/kata1.go:14]: is this not an edge case where start == end == target?
